frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect } from 'react'
+import { Box, Flex, FlexProps, Image, Stack, Text } from '@chakra-ui/react'
+import { format } from 'date-fns'
+
+import { BxsChevronUp } from '~assets/icons/BxsChevronUp'
+import Button from '~components/Button'
+
+import { useAdminForm } from '~features/admin-form/common/queries'
+import { ThankYouSvgr } from '~features/public-form/components/FormEndPage/components/ThankYouSvgr'
+
+import {
+  endPageDataSelector,
+  setStateSelector,
+  useEndPageBuilderStore,
+} from '../useEndPageBuilderStore'
+
+export const EndPageView = ({ ...props }: FlexProps): JSX.Element => {
+  const { data: form } = useAdminForm()
+  const { endPageData, setState } = useEndPageBuilderStore((state) => ({
+    endPageData: endPageDataSelector(state),
+    setState: setStateSelector(state),
+  }))
+
+  useEffect(() => {
+    if (form) setState(form.endPage)
+  }, [form, setState])
```

Comment:
same thing as what we were talking about earlier - the old store state from a different form gets loaded first, before getting updated to the correct form state

Reply:
feel like this is just fixing the symptom instead of the cause. like discussed, maybe we should reset the store on unmount instead of setting it on mount.

```tsx
useEffect(() => {
  return () => {
    resetStore()
  }
}, [])
```